explorer/src/pages/TransactionDetailsPage.tsx

```diff
@@ -75,14 +82,54 @@ function StatusCard({ signature }: Props) {
     [fetchStatus, fetchDetails]
   );
 
+  React.useEffect(() => {
+    return () => {
+      if (autoRefreshTimeout.current) {
+        clearTimeout(autoRefreshTimeout.current);
+      }
+    };
+  }, []);
+
   // Fetch transaction on load
   React.useEffect(() => {
     if (!status && clusterStatus === ClusterStatus.Connected) {
       fetchStatus(signature);
     }
   }, [signature, clusterStatus]); // eslint-disable-line react-hooks/exhaustive-deps
 
-  if (!status || status.status === FetchStatus.Fetching) {
+  React.useEffect(() => {
+    if (!status || !status.data?.info) {
+      return;
+    }
+
+    if (status.status !== FetchStatus.Fetching && autoRefreshInProcess) {
+      setAutoRefreshInProcess(false);
```

Comment:
It feels to me that the `autoRefreshInProcess` state should be very closely tied to whether there is an active timer or not. Could you try moving this to where you call `clearTimeout` below? Or is the intent to intermittently toggle the state to indicate to the user that it's retrying again?

Reply:
Yeah, the original intent was it to intermittently toggle the state to indicate to the user, but I think you're right about the state reflecting active timeout, and I'm going to use an interval instead!